bb/src/main/java/com/richardmcdougall/bb/BoardVisualization.java

```diff
@@ -284,12 +293,41 @@ public int displayAlgorithm(String algorithm) {
                 mVisualizationMickey.update(Mickey.kMickeyBlank);
                 break;
 
+            // JosPack visualizations go here
+            case "JPGold()":
+                frameRate = mFrameRate;
+                mVisualizationJosPack.update(JosPack.kJPGold);
+                break;
+
+            case "JPSparkle()":
+                frameRate = mFrameRate;
+                mVisualizationJosPack.update(JosPack.kJPSparkle);
+                break;
+
+            case "JPColors()":
+                frameRate = mFrameRate;
+                mVisualizationJosPack.update(JosPack.kJPColors);
+                break;
+
+            case "JPBlueGold()":
+                frameRate = mFrameRate;
+                mVisualizationJosPack.update(JosPack.kJPBlueGold);
+                break;
+
+            case "modeJPBlank()":
+                frameRate = mFrameRate;
+                mVisualizationJosPack.update(JosPack.kJPBlank);
+                break;
+            // End JosPack visualizations
+
             case "modePlayaMap()":
                 frameRate = mFrameRate;
                 mVisualizationPlayaMap.update(Visualization.kDefault);
                 break;
 
             default:
+                Log.d(TAG, "Could not find visualization algorithm: " + algorithm);
```

Comment:
We can't log.d every frame -- if we ever hit it it causes hang side effects on android. Maybe just print once?

Reply:
Fair enough - not sure how to print only once without keeping a status bit, so for now, we'll leave it commented out.